src/common/gui/base/colorPicker/ColorPickerView.ts

```diff
@@ -0,0 +1,373 @@
+import m, { Children, Component, Vnode } from "mithril"
+import { px, size } from "../../size.js"
+import { ExpanderButton, ExpanderPanel } from "../Expander.js"
+import stream from "mithril/stream"
+import { TextField } from "../TextField.js"
+import { lang } from "../../../misc/LanguageViewModel.js"
+import { hexToHSL, hslToHex, isValidCSSHexColor, MAX_HUE_ANGLE, normalizeHueAngle } from "../Color.js"
+import { ColorPickerModel } from "./ColorPickerModel.js"
+import { client } from "../../../misc/ClientDetector.js"
+import { theme } from "../../theme.js"
+
+const enum Sizes {
+	hueGradientBorderWidth = 1,
+	hueGradientHeight = 40,
+}
+
+const enum PaletteIndex {
+	defaultVariant = 2,
+	customVariant = -1,
+}
+
+type ColorPickerAttrs = {
+	value: string
+	onselect: (color: string) => void
+}
+
+export class ColorPickerView extends ColorPickerModel implements Component<ColorPickerAttrs> {
```

Comment:
we try to avoid inheritance in favor of composition, especially for UI components. I am also not sure why it was done in this case as all parts of `ColorPickerModel` are static.